backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepositoryCache.java

```diff
@@ -36,7 +36,7 @@ public IGitMacheteRepository getInstance(Path rootDirectoryPath, Path mainGitDir
     val key = Tuple.of(rootDirectoryPath, worktreeGitDirectoryPath);
     val valueReferenceOption = gitMacheteRepositoryCache.get(key);
```

Comment:
Nit: while you're here, just convert to `@Nullable` (some `getOrNull()` or sth) and then check `!= null`